### PR TITLE
Reorganization in support of preparing for mono-repo

### PR DESCRIPTION
**Pull request by treubig26**
Moves the armory-library source into the `library` sub-folder. Additionally, the `charmory_examples` package is renamed to `armory/examples` where `armory` becomes a [namespace package](https://packaging.python.org/en/latest/guides/packaging-namespace-packages/).

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#
```
Comment (mwartell) [inline requirements.txt:1]: This requirements.txt has got to go: it is semantically overloaded

```
$ cat install.sh
#!/bin/bash
echo "installing all armory-suite components with pip"
set -x -e
pip install .
pip install library
pip install examples[all]
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -75,10 +48,12 @@ jobs:
           python-version: ${{ matrix.python-version }}
 
       - name: Install
-        run: make install
+        run: |
+          pip3 install --no-compile --editable .
```
Comment (mwartell) [inline .github/workflows/ci.yml:52]: There is extreme overuse of editable in Armory (new and old) materials. Most people aren't developing Armory so we should never provide `--editable` without really good cause.

In CI, `--editable` is non-sensical unless you are testing editable

File: pyproject.toml [hunk 3]
```diff
@@ -171,15 +59,15 @@ markers = [
...
-#     ignore::Warning:art*
+
 
 [tool.isort]
 profile = "black"
 force_sort_within_sections = true
+src_paths = ["library/src"]
+
+
+[tool.taskipy.tasks]
```
Comment (mwartell) [inline pyproject.toml:70]: I like taskipy, so fresh, so clean.

**Review by mwartell — CHANGES_REQUESTED** [review]
Sample replacement code for requirements.txt included.

**Review by mwartell — CHANGES_REQUESTED** [review]
Sample replacement code for requirements.txt included.

File: dev-install.sh [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+fi
+
+set +e
+PYTHON=$(which ${PYTHON:-python})
+set -e
+
+if [ -z "$PYTHON" ]; then
+    echo "Python interpreter not found. Exiting."
+    exit 1
+fi
```
[the remark this answers is not in the document]
Comment (treubig26) [reply dev-install.sh:20]: I'm not sure I follow. Are you suggesting something different than the echo on line 22?

**Pull request merged**